### DISCUSSION TITLE
Webpack resolve alias with env var

### DISCUSSION THREAD

**Discussion by lucalategola** [vercel/next.js#69448] [Help]
### Summary

My app returns the error 'Cannot find module '@tenant/foo' or its corresponding type declarations.ts(2307)' in a typescript file when defining a dynamic alias resolve via webpack.
I defined a webpack resolve alias dinamically with an env var like this in next.config.mjs:
```javascript
webpack(config) {
  config.resolve.alias = {
    ...config.resolve.alias,
    '@tenant': path.resolve(__dirname, `src/tenant/${process.env.TENANT}/`),
  };

  return config;
},
```
And in my tsconfig.json I added the tenant entry for the path:
```javascript
"paths": {
  "@/*": ["./src/*"],
  "@tenant/*": ["./src/tenant/*"],
}
```
I import my foo function into a page component with `import { fooFunction} from '@tenant/foo';` and the function it works (the console log output the log), but vscode give me this error `Cannot find module '@tenant/foo' or its corresponding type declarations.ts(2307)`.

Anyone that could help me?

Thanks

**Comment by sambecker** [comment]
Did you ever figure this out @lucalategola? Trying to accomplish something similar with an env var-based import strategy.
